libclc/clc/lib/generic/math/clc_fmax.cl

```diff
@@ -10,13 +10,29 @@
 #include <clc/internal/clc.h>
 #include <clc/relational/clc_isnan.h>
 
-_CLC_DEFINE_BINARY_BUILTIN(float, __clc_fmax, __builtin_fmaxf, float, float);
+#define __FLOAT_ONLY
+#define __CLC_MIN_VECSIZE 1
+#define FUNCTION __clc_fmax
+#define __IMPL_FUNCTION __builtin_fmaxf
```

Comment:
Perhaps this isn't the best use-case of `binary_def_scalarize.inc`. Could we instead use `binary_def.inc` with `FUNCTION = __clc_max` and `__CLC_FUNCTION(x) __builtin_fmaxf`? That way we're not setting `__CLC_MIN_VECSIZE 1`.

As an aside I think the name `__CLC_FUNCTION` could be renamed `IMPL_FUNCTION` as in these new "scalarize" files, but that's a later refactor.